lib/amperize.js

```diff
@@ -170,46 +171,60 @@ Amperize.prototype.traverse = function traverse(data, html, done) {
      * @return {Object} element incl. width and height
      */
     function getImageSize(element) {
-      var options = url.parse(element.attribs.src),
-          timeout = 5000,
-          request = element.attribs.src.indexOf('https') === 0 ? https : http;
+      var imagePath = url.parse(element.attribs.src),
+          requestOptions,
+          timeout = 5000;
 
       called = false;
 
+      if (!validator.isURL(imagePath.href)) {
+        if (called) return;
```

Comment:
Why do we have to keep the `called` variable?
I see three obvious cases:

1. invalid url
2. success
3. error

Can you please explain?

Reply:
I wasn't sure if this is still needed when switching to `got`. I removed it now, as `got` should be able to handle it properly.